app.py

```diff
@@ -103,6 +120,66 @@ def simple_flow(times):
 
     my_bulb.start_flow(flow1)
 
+def split_times_to_datetimes(split_times):
+    results = []
+    current_time = datetime.datetime.today()
+    for time in split_times:
+        test_time = str(current_time.year) + " " + str(current_time.month)+ " " + str(current_time.day)+ " " + str(time[0]) + str(time[1])
+        dateobj = datetime.datetime.strptime(test_time, '%Y %m %d %I:%M %p')
+        results.append([dateobj, time[2].split(' ')[0]])
+
+    return results
+
+def get_data_points(current_time, tide_times):
+    results = []
+    prev_time = None
+    next_time = None
+    last_item = None
+    #Compare current time with tide times
+    for time in tide_times:
+        if time[0] < current_time:
+            last_item = time[0]
+        else:
+            prev_time = last_item
+            next_time = time[0]
+            results.extend(prev_time, current_time, next_time)
+    return results
+
+def get_light_intensity(data_points):
+    # Get fraction of tide
+    # light_intensity = time.mktime(data_points[0][0].timetuple())
+    prev_time = data_points[0]
+    current_time = data_points[1]
+    next_time = data_points[2]
+
+    prev_time_seconds = float(time.mktime(prev_time.timetuple()))
+    current_time_seconds = float(time.mktime(current_time.timetuple()))
+    next_time_seconds = float(time.mktime(next_time.timetuple()))
+
+    numerator = (next_time_seconds - prev_time_seconds)/10
+    denomenator = current_time_seconds - prev_time_seconds
+    light_intensity = (numerator/denomenator) * 100
+
+    return light_intensity
```

Comment:
This can certainly be used with the flow function as a means to adjust brightness. We can run a thread that just infinitely loops this function. We can think about if we want to put the 'high / low notification' within said loop.  Or run a separate delayed thread.